File: sandbox-runner/patterns/delayed-callback/java/src/main/java/Shared.java
public final class Shared {
    public static final String TASK_QUEUE = "webhooks-task-queue";
    public static final String WORKFLOW_ID_PREFIX = "order";
    public static final String CALLBACK_WORKFLOW_ID_PREFIX = "delayed-callback";

    public record OrderInput(String orderId, double amount) {}

    public record PaymentPayload(String paymentId, double amount) {}

    public record CallbackInput(String callbackUrl, String payload, int delaySeconds) {}

    private Shared() {}
}

File: sandbox-runner/patterns/delayed-callback/go/shared.go
package main

const (
	TaskQueue               = "webhooks-task-queue"
	WorkflowIDPrefix        = "order"
	CallbackWorkflowIDPrefix = "delayed-callback"
	SignalName              = "payment_received"
)

type OrderInput struct {
	OrderID string  `json:"order_id"`
	Amount  float64 `json:"amount"`
}

type PaymentPayload struct {
	PaymentID string  `json:"payment_id"`
	Amount    float64 `json:"amount"`
}

type CallbackInput struct {
	CallbackURL  string `json:"callback_url"`
	Payload      string `json:"payload"`
	DelaySeconds int    `json:"delay_seconds"`
}
